app/src/main/java/net/christianbeier/droidvnc_ng/MainService.java

```diff
@@ -770,29 +859,45 @@ static ArrayList<String> getIPv4s() {
         }
 
         // not running on Chrome OS
+        String listenInterface = MainService.getListenInterface();
+
         try {
-            // thanks go to https://stackoverflow.com/a/20103869/361413
-            Enumeration<NetworkInterface> nis = NetworkInterface.getNetworkInterfaces();
-            NetworkInterface ni;
-            while (nis.hasMoreElements()) {
-                ni = nis.nextElement();
-                if (!ni.isLoopback()/*not loopback*/ && ni.isUp()/*it works now*/) {
-                    for (InterfaceAddress ia : ni.getInterfaceAddresses()) {
-                        //filter for ipv4/ipv6
-                        if (ia.getAddress().getAddress().length == 4) {
-                            //4 for ipv4, 16 for ipv6
-                            hosts.add(ia.getAddress().toString().replaceAll("/", ""));
+            if (listenInterface.equals("0.0.0.0")) {
+                // Any mode: get all the available NICs and add their IPv4
+                ArrayList<NetworkInterface> nics = MainService.getAvailableNICs();
+                for (NetworkInterface nic : nics) {
+                    if (!nic.isLoopback()) {
+                        ArrayList<String> ipv4s = MainService.getIPv4ForInterface(nic);
+                        for (String ipv4 : ipv4s) {
+                            hosts.add(ipv4);
                         }
                     }
                 }
+            } else {
+                // Single interface: get all its IPv4 addresses
+                ArrayList<String> ipv4s = MainService.getIPv4ForInterface(NetworkInterface.getByName(listenInterface));
+                for (String ipv4 : ipv4s) {
+                    hosts.add(ipv4);
+                }
             }
-        } catch (SocketException e) {
-            //unused
+        } catch (SocketException ex) {
+            // unused
         }
 
         return hosts;
     }
 
+
+
+    static String getListenInterface() {
+        try {
+            SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(instance);
+            return prefs.getString(PREFS_KEY_SERVER_LAST_LISTEN_INTERFACE, new Defaults(instance).getListenInterface());
+        } catch (Exception e) {
+            return null;
```

Comment:
Rather "0.0.0.0"? Unsure...

Reply:
What do you suggest?

Reply:
Changed default interface to loopback, this way, it is safer.